### PR TITLE
Averaged planes executor

### PR DESCRIPTION
**Pull request by gyalla**
Adding an executor for averaging planes with actions to compute the rotor averaged velocity and plot averaged planes. 

Additionally, I have fixed a bug related to how actions get executed and updated the instantaneous executor as well. 

Ideally, this executor could take a list of netcdf files and average over all of them and combine the results. Is there a way to do this in the frontend?

### CODE REVIEW

**Comment by lawrenceccheung** [comment]
hi @gyalla, 

I made a change recently to `avgPlaneXR()` in `postproamrwindsample_xarray.py` so that it could take a list of netcdf files and average over all of them in the correct fashion:
https://github.com/Exawind/amr-wind-frontend/blob/ff4febd8bb3612a2138f9e2005bce52adbd5bf05/postproamrwindsample_xarray.py#L195-L208
(This assumes the list of files is given in the correct sequential order.)  I think there's nothing you need to do, if you provide a list of files as input to `ncfile`, it should work out in this executor.

Lawrence

File: postproengine/averageplanes.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+            print('Executing '+self.actionname)
+            yc = self.actiondict['yc']
+            zc = self.actiondict['zc']
+            Radius =  self.actiondict['Diam']/2.0
+            iplanes = self.actiondict['iplane']
+            savefile = self.actiondict['savefile']
+            rotor_avg = {}
+            xloc = {}
+            if not isinstance(iplanes, list): iplanes = [iplanes,]
+            for iplane in iplanes:
```
Comment (lawrenceccheung) [inline postproengine/averageplanes.py:135]: This assumes that the rotor planes are Y-Z planes.  There's no problem with that now, later on we might generalize this to handle other wind turbine orientations.

File: postproengine/averageplanes.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+            pklfile  = plane['savepklfile']
+            varnames = plane['varnames']
+            #Get all times if not specified 
+            if tavg==[]:
+                timevec = ppsample.getVar(ppsample.loadDataset(ncfile), 'time')
+                tavg = [timevec[0].data,timevec[-1].data]
+                print("No time interval specified. Averaging over entire file: ",tavg)
+
+            # Load the plane
+            self.dbavg  = ppsamplexr.avgPlaneXR(ncfile, tavg, varnames=varnames, groupname=group, savepklfile=pklfile, verbose=verbose)
```
Comment (lawrenceccheung) [inline postproengine/averageplanes.py]: We could add a flag here to load `dbavg` from the pklfile instead of recomputing it.

**Comment by gyalla** [comment]
Thanks @lawrenceccheung, that's great that the functionality is already built into the frontend. I updated the average planes executor to make sure it would interface with one or multiple netcdf files accordingly.

**Review by lawrenceccheung — APPROVED** [review]

**Pull request merged**